### PR TITLE
Add in before_send config option to sentry integration

### PR DESCRIPTION
**Pull request by leonsmith**
Adds in the ability to define a before_send callback to the sentry SDK

https://docs.sentry.io/platforms/python/configuration/filtering/

### CODE REVIEW

File: airflow/sentry.py [hunk 1]
```diff
@@ -107,6 +107,9 @@ def __init__(self):
                         ", ".join(unsupported_options),
                     )
 
+                if sentry_config_opts.get('before_send'):
+                    sentry_config_opts['before_send'] = import_string(sentry_config_opts['before_send'])
```
Comment (mik-laj) [inline airflow/sentry.py]: Can you use `conf.getimport`? Thanks to this, we will have a more friendly error message.

Comment (leonsmith) [reply airflow/sentry.py]: Good suggestion, switched it to use getimport.

**Comment by leonsmith** [comment]
I'm unsure how to get the patch to work with the pytest fixture. Some extra eyeballs & suggestions would be appreciated

**Comment by mik-laj** [comment]
@leonsmith  Can you do a rebase? This problem should now be fixed on the main branch

**Comment by mik-laj** [comment]
Static checks failed. Can you fix it?

**Comment by leonsmith** [comment]
This is all passing now, test failure is a flaky one that's present on main

**Review by mik-laj — APPROVED** [review]

**Pull request merged**